app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class UsersController < API::V1::ApplicationController
+      include API::V1::ErrorHandlerConcern
+
+      skip_before_action :doorkeeper_authorize!, only: :create
+
+      before_action :ensure_valid_client, only: :create
+
+      def create
+        user = User.new(create_params.except(:client_id))
+
+        render_errors ActiveModel::ErrorsSerializer.new(user.errors).serializable_hash and return unless user.save
+
+        render json: UserTokenSerializer.new(user, { params: { client_id: @client_app.id } }).serializable_hash,
+               status: :created
+      end
+
+      private
+
+      def create_params
+        params.permit(:email, :password, :last_name, :first_name, :client_id)
+      end
+
+      def ensure_valid_client
+        @client_app = Doorkeeper::Application.find_by(uid: create_params[:client_id])
```

Comment:
I am not sure only validating `client_id` is the best practice or not. Its seems it is better to validate both `client_id`, `secret` for better security .
```suggestion
        @client_app = Doorkeeper::Application.by_uid_and_secret(params[:client_id], params[:client_secret])
```

In the Survey api, it required both for registration also. Wjat do you think?

Reply:
Makes good sense~ Fixed ;-)